router/router.nginx.conf.in

```diff
@@ -0,0 +1,211 @@
+
+        proxy_intercept_errors on;
+        # 502: no service, 504: pod not responding
+        error_page 502 504 = @pod_dead;
+    }
+
+    location @pod_dead {
+        return 307 $scheme://$http_host/new;
+    }
+
+    location ~ /instance-ready/([^/]+)/ {
+        resolver kube-dns.kube-system.svc.cluster.local;
+        proxy_pass http://$1.default.svc.cluster.local/instance/$1/;
+    }
+
+    location / {
+        proxy_pass http://notebook/;
+
+        proxy_set_header Host $http_host;
+        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+        proxy_set_header X-Forwarded-Host $http_host;
+        proxy_set_header X-Forwarded-Proto $scheme;
+        proxy_set_header X-Real-IP $remote_addr;
+        proxy_http_version 1.1;
+        proxy_redirect off;
+        proxy_buffering off;
+        proxy_set_header Upgrade $http_upgrade;
+        proxy_set_header Connection "upgrade";
+        proxy_read_timeout 86400;
+    }
+
+    listen 80;
+    listen [::]:80;
+}
+
+server {
+    server_name notebook2.@domain@;
+
+    # needed to correctly handle error_page with internal handles
+    recursive_error_pages on;
+
+    location = /auth {
+        internal;
+        proxy_pass http://notebook2/auth/$pod_uuid;
+    }
+
+    location ~ /instance/([^/]+)/(.*) {
+        set $pod_uuid $1;
+        auth_request /auth;
+        auth_request_set $auth_pod_ip $upstream_http_pod_ip;
+
+        proxy_pass http://$auth_pod_ip$request_uri;
+
+        proxy_set_header Host $http_host;
+        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+        proxy_set_header X-Forwarded-Host $http_host;
+        proxy_set_header X-Forwarded-Proto $scheme;
+        proxy_set_header X-Real-IP $remote_addr;
+        proxy_http_version 1.1;
+        proxy_redirect off;
+        proxy_buffering off;
+        proxy_set_header Upgrade $http_upgrade;
+        proxy_set_header Connection "upgrade";
+        proxy_read_timeout 86400;
+
+        proxy_connect_timeout 5s;
+
+        proxy_intercept_errors on;
+        # 502: no service, 504: pod not responding
+        error_page 502 504 = @pod_dead;
+    }
+
+    location @pod_dead {
+        set $args "?err=Notebook not found";
+        return 307 $scheme://$http_host/error$args;
+    }
+
+    location ~ /instance-ready/([^/]+)/ {
+        set $pod_uuid $1;
+        auth_request /auth;
+        auth_request_set $auth_pod_ip $upstream_http_pod_ip;
+
+        proxy_pass http://$auth_pod_ip/instance/$1/;
+    }
+
+    location / {
+        proxy_pass http://notebook2/;
+
+        proxy_set_header Host $http_host;
+        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+        proxy_set_header X-Forwarded-Host $http_host;
+        proxy_set_header X-Forwarded-Proto $scheme;
+        proxy_set_header X-Real-IP $remote_addr;
+        proxy_http_version 1.1;
+        proxy_redirect off;
+        proxy_buffering off;
+        proxy_set_header Upgrade $http_upgrade;
```

Comment:
Is all of this a copy from somewhere?

Reply:
Yes, from `gateway/hail.nginx.conf.in`.  Basically, it is that file, minus the letsencrypt block (which only the gateway needs to handle) and all blocks switched to listen on http instead of https.